### PR TITLE
PP-4779 extra tests and edge case handling

### PR DESCRIPTION
**Pull request by gidsg**
## WHAT
Extra test and functionality for PP-4779

Given a charge exists in our system
And the account associated with the charge has no gateway merchant id
When I call
GET /v1/frontend/charges/{chargeId}
Then the gateway_merchant_id is null

Given I have a worldpay merchant id for a service
And the service exists in our system
And the service has no gateway account credentials
When I PATCH the worldpay merchant id
Then a 400 BAD REQUEST is returned

## HOW 
See user stories above

### CODE REVIEW

**Comment by gidsg** [comment]
@oswaldquek I think this is ready for review now. I believe the Codacy warnings are a false-positive as `rest-assured` wraps the assertions in the tests.

File: src/main/java/uk/gov/pay/connector/gatewayaccount/exception/MerchantIdWithoutCredentialsException.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.pay.connector.gatewayaccount.exception;
+
+import uk.gov.pay.connector.util.ResponseUtil;
+
+import javax.ws.rs.BadRequestException;
+
+public class MerchantIdWithoutCredentialsException extends BadRequestException {
```
Comment (oswaldquek) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/exception/MerchantIdWithoutCredentialsException.java:7]: normally we do `implements ExceptionMapper` (see ValidationExceptionMapper and UnsupportedOperationExceptionMapper for example) but it seems InvalidKeyException extends BadRequestException too so there doesn't seem to be any standard for this. (This isn't a request to change it, just an observation.)

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceAllowWebPaymentsITest.java [hunk 2]
```diff
@@ -79,6 +79,20 @@ public void assertGooglePayPermission(boolean setAllowGooglePayFlag, boolean set
...
+    public void assertBadRequestResponseIfPatchingMerchantIdWithNoGatewayAccountCredentials() throws JsonProcessingException {
+        String accountIdWithoutGatewayAccountCredentials = extractGatewayAccountId(createAGatewayAccountFor(testContext.getPort(), "worldpay"));
+        String payload = new ObjectMapper().writeValueAsString(ImmutableMap.of("op", "add",
+                "path", "credentials/gateway_merchant_id",
+                "value", "1234abc"));
+        given().port(testContext.getPort()).contentType(JSON)
+                .body(payload)
+                .patch("/v1/api/accounts/" + accountIdWithoutGatewayAccountCredentials)
+                .then()
+                .assertThat()
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceAllowWebPaymentsITest.java]: you don't need this `assertThat` line

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceAllowWebPaymentsITest.java [hunk 3]
```diff
@@ -79,6 +79,20 @@ public void assertGooglePayPermission(boolean setAllowGooglePayFlag, boolean set
...
+                .statusCode(HttpStatus.SC_BAD_REQUEST);
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceAllowWebPaymentsITest.java:95]: could you also assert the error message `"Account Credentials are required to set a Gateway Merchant ID"`? an easy way of printing the actual returned json body is to use `.log().body()` after the `.then()` part. 

i think the assertion will be `.path("message", is("Account Credentials are required to set a Gateway Merchant ID"))`

**Review by oswaldquek — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/pay/connector/gatewayaccount/exception/MerchantIdWithoutCredentialsException.java (hunk 1 above)
Comment (gidsg) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/exception/MerchantIdWithoutCredentialsException.java:7]: OK, I took this style from https://github.com/alphagov/pay-connector/blob/a4f6a67778691ff82fc907d2b7830dfdc2aa8ff6/src/main/java/uk/gov/pay/connector/wallets/applepay/InvalidKeyException.java

**Review by oswaldquek — APPROVED** [review]
